### PR TITLE
Implement ErrorBody as the default errorsome response object

### PR DESCRIPTION
**Pull request by dhartwich1991**
_Implement Error Body as the default response object when a request was not executed with the status code 200-299._
- Added the ErrorBody class
- Set the ErrorBody as default errorType in the CallSpec
- Kept the errorAs builder method for internal use
- Replaced Object or String with ErrorType as the error type in all resources that have been implemented already

Dependencies:
**Reviewer**:
- [x] Allowed to merge
- [x] @serj-lotutovici

### CODE REVIEW

File: sdk/src/main/java/com/xing/android/sdk/ErrorBody.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    public String getErrorMessage() {
+        return errorMessage;
+    }
+
+    public List<Error> getErrors() {
+        return errors;
+    }
+
+    private static final class Error {
```
Comment (serj-lotutovici) [inline sdk/src/main/java/com/xing/android/sdk/ErrorBody.java]: This class should be `public`

File: sdk/src/main/java/com/xing/android/sdk/ErrorBody.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        ErrorBody that = (ErrorBody) o;
+
+        return (errorName != null ? errorName.equals(that.errorName) : that.errorName == null) && (errorMessage != null
```
Comment (serj-lotutovici) [inline sdk/src/main/java/com/xing/android/sdk/ErrorBody.java]: please reformat. so that every `&&` in on a new line.

**Comment by serj-lotutovici** [comment]
Aside form my comments looks awesome.

**Comment by dhartwich1991** [comment]
Applied the changes from your review :+1:

**Pull request merged**